tensorboard/webapp/widgets/histogram/histogram_v2_component.ts

```diff
@@ -239,8 +231,25 @@ export class HistogramV2Component
 
   onResize() {
     this.updateClientRects();
-    this.updateChart();
-    this.changeDetector.detectChanges();
+    this.updateChartIfVisible();
+  }
+
+  onVisibilityChange({visible}: {visible: boolean}) {
+    this.domVisible = visible;
+    if (!visible) return;
+
+    if (!this.domInitialized) {
+      this.domInitialized = true;
+      this.changeDetector.detectChanges();
```

Comment:
Is the ordering of these lines intentionally set to put detectChanges before calling `updateClientRects` below?  If so, could we add a comment?